projects/public-search/src/app/api/loan-api.service.ts

```diff
@@ -0,0 +1,166 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Record, RecordService } from '@rero/ng-core';
+import { Error } from '@rero/ng-core/lib/error/error';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { BaseApi } from './base-api';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoanApiService extends BaseApi {
+
+  /**
+   * Constructor
+   * @param _recordService - RecordService
+   * @param _httpClient - HttpClient
+   */
+  constructor(
+    private _recordService: RecordService,
+    private _httpClient: HttpClient
+  ) {
+    super();
+  }
+
+  /**
+   * Get loan
+   * @param patronPid - string
+   * @param page - number
+   * @param itemsPerPage - number
+   * @param headers - object
+   * @return Observable
+   */
+  getOnLoan(
+    patronPid: string, page: number,
+    itemsPerPage: number = 10, headers = BaseApi.reroJsonheaders
+  ): Observable<Record | Error> {
+    const loanStates = ['ITEM_ON_LOAN'];
+    return this._recordService.getRecords(
+      'loans', this._patronStateQuery(patronPid, loanStates), page, itemsPerPage,
+      undefined, undefined, headers
+    );
+  }
+
+  /**
+   * Get Request
+   * @param patronPid - string
+   * @param page - number
+   * @param itemsPerPage - number
+   * @param withHeaders - boolean
+   * @return Observable
+   */
+  getRequest(
+    patronPid: string, page: number,
+    itemsPerPage: number = 10, headers = BaseApi.reroJsonheaders
+  ): Observable<Record | Error> {
+    const requestStates = ['PENDING', 'ITEM_AT_DESK', 'ITEM_IN_TRANSIT_FOR_PICKUP'];
+    return this._recordService.getRecords(
+      'loans', this._patronStateQuery(patronPid, requestStates), page, itemsPerPage,
+      undefined, undefined, headers
+    );
+  }
+
+  /**
+   * Get history
+   * @param patronPid - string
+   * @param page - number
+   * @param itemsPerPage - number
+   * @param headers - object
+   * @return Observable
+   */
+  getHistory(
+    patronPid: string, page: number,
+    itemsPerPage: number = 10, headers = BaseApi.reroJsonheaders
+  ): Observable<Record | Error> {
+    const historyStates = ['ITEM_RETURNED', 'ITEM_IN_TRANSIT_TO_HOUSE', 'CANCELLED'];
+    const queryHistory = [
+      this._patronStateQuery(patronPid, historyStates),
+      '_exists_:end_date'
+    ].join(' AND ');
+    return this._recordService.getRecords(
+      'loans', queryHistory, page, itemsPerPage,
+      undefined, undefined, headers
+    );
+  }
+
+  /**
+   * Is loan can extend
+   * @param loanPid - string
+   * @returns Observable
+   */
+  canExtend(loanPid: string): Observable<CanExtend> {
+    return this._httpClient.get(`/api/loan/${loanPid}/can_extend`)
+      .pipe(
+        catchError(() => of(undefined)),
+        map((response: any) => {
+          return response;
+        })
```

Comment:
Is is necessary ? you get `response` and you return `response`. 
Cosmetic : as it's a single line statement, you can just use `map((response: any) => response)`

Reply:
done